tests/test_request_media.py

```diff
@@ -199,17 +211,138 @@ def test_complete_consumption(asgi):
+    res = client.simulate_get('/', body=b'{' if body else '')
+    if body:
+        assert res.status_code == 400
+        assert 'Could not parse JSON body' in res.json['description']
+    else:
+        assert res.status_code == 200
+
+
+def test_fallback_does_not_override_media_default(asgi):
+    client = create_client(asgi, resource=FallBackAsync() if asgi else FallBack())
+
+    res = client.simulate_get('/', headers={'Content-Type': 'application/x-www-form-urlencoded'})
+    assert res.status_code == 200
+    assert res.text == '{}'
+
+
+async def _check_error(req, isasync):
+    err, err2 = None, None
+    try:
+        (await req.media) if isasync else req.media
+    except Exception as e:
+        err = e
+    try:
+        (await req.get_media()) if isasync else req.get_media()
+    except Exception as e:
+        err2 = e
+    assert err is not None
+    assert err2 is not None
+    assert err2 is err
+    if isinstance(err, errors.MediaMalformedError):
+        assert err2.__cause__ is err.__cause__
+    obj = {}
+    if req.get_param_as_bool('empty'):
+        # default_when_empty is not cached
+        res = (await req.get_media(obj)) if isasync else req.get_media(obj)
+        assert res is obj
+        err3 = None
+        try:
+            ((await req.media) if isasync else req.media) is obj
+        except Exception as e:
+            err3 = e
+        assert err3 is err
+    else:
+        err3 = None
+        try:
+            (await req.get_media(obj)) if isasync else req.get_media(obj)
+        except Exception as e:
+            err3 = e
+        assert err3 is err
+    raise errors.HTTPError(falcon.HTTP_IM_A_TEAPOT)
+
+
+class RepeatedError:
+    def on_get(self, req, resp):
+        util.async_to_sync(_check_error, req, False)
+
+
+class RepeatedErrorAsync:
+    async def on_get(self, req, resp):
+        await _check_error(req, True)
+
+
+@pytest.mark.parametrize('body', ('{', ''))
+def test_repeated_error(asgi, body):
+    client = create_client(asgi, resource=RepeatedErrorAsync() if asgi else RepeatedError())
+
+    res = client.simulate_get('/', body=body, params={'empty': not bool(body)})
+    assert res.status == falcon.HTTP_IM_A_TEAPOT
+
+
+def test_error_after_first_default(asgi):
+    async def _check_error(req, isasync):
+        assert (await req.get_media(42)) if isasync else req.get_media(42) == 42
+        try:
+            (await req.get_media()) if isasync else req.get_media()
+        except falcon.MediaNotFoundError:
+            raise falcon.HTTPStatus(falcon.HTTP_749)
+        raise falcon.HTTPStatus(falcon.HTTP_703)
+
+    class Res:
+        def on_get(self, req, resp):
+            util.async_to_sync(_check_error, req, False)
+
+    class ResAsync:
+        async def on_get(self, req, resp):
+            await _check_error(req, True)
+
+    client = create_client(asgi, resource=ResAsync() if asgi else Res())
+
+    res = client.simulate_get('/', body='')
+    assert res.status == falcon.HTTP_749
```

Comment:
This is probably fine as-is, but note that `HTTP 749` is very unlikely to work on a real ASGI server (in ASGI, the response status is provided as an integer status code).

Reply:
that makes me sad 😂

Reply:
Without a doubt, that is a glaring deficiency in that spec.